### PR TITLE
Fix multiple pending upgrades bug, remove node cloud status PUT

### LINKED ISSUE #68: Mass upgrading from 1.12.x to 1.13.x not completing
#### What is the current behavior?
upgrading from 1.12 to 1.13 is hanging

#### What is the expected behavior?
it upgrades without issue

### Reproduction
1. Create a cluster with Kube v. 1.11.x
* Provider: DO
* Region: NYC 1
* Master Pools: 3 with 1 node each
* Worker Pools: 3 with 1 node each
2. Mass upgrade of 1.11x to 1.12.x of `master` pools, **successful**
3. Mass upgrade of 1.11x to 1.12.x of `worker` pools, **successful**
4. Mass upgrade of 1.12x to 1.13.x of `master` pools, **hanging**

### Evidence
Org: `containership-dev`, clusters `upgrading` & `upgrade-2`

### PR DESCRIPTION
**Pull request by mattkelly**
#### What does this pull request accomplish?

 - Fix bug where already-complete upgrade could go next
 - Remove node cloud status update

See commit messages for details.

 #### What issue(s) does this fix?
* Fixes #68

 #### Additional Considerations

 ### Testing

 Same process as outlined in #68

 ### Dependencies
 None

### CODE REVIEW

File: pkg/coordinator/upgrade_controller.go [hunk 1]
```diff
@@ -595,6 +578,17 @@ func getFinalUpgradeStatus(cup *provisioncsv3.ClusterUpgrade) provisioncsv3.Upgr
 	return provisioncsv3.UpgradeSuccess
 }
 
+// isUpgradePending returns true if an upgrade has not been processed yet
+func isUpgradePending(cup *provisioncsv3.ClusterUpgrade) bool {
+	// ClusterUpgrades to not have a pending status; the empty status is
+	// recognized as pending.
+	if cup.Spec.Status.ClusterStatus == "" {
```
Comment (mattkelly) [inline pkg/coordinator/upgrade_controller.go]: Collapse into a single line

File: pkg/coordinator/upgrade_controller.go [hunk 2]
```diff
@@ -595,6 +578,13 @@ func getFinalUpgradeStatus(cup *provisioncsv3.ClusterUpgrade) provisioncsv3.Upgr
 	return provisioncsv3.UpgradeSuccess
 }
 
+// isUpgradePending returns true if an upgrade has not been processed yet
+func isUpgradePending(cup *provisioncsv3.ClusterUpgrade) bool {
+	// ClusterUpgrades to not have a pending status; the empty status is
```
Comment (ashleyschuett) [inline pkg/coordinator/upgrade_controller.go]: do not?

File: pkg/coordinator/upgrade_controller.go [hunk 3]
```diff
@@ -595,6 +578,13 @@ func getFinalUpgradeStatus(cup *provisioncsv3.ClusterUpgrade) provisioncsv3.Upgr
...
+	// recognized as pending.
+	return cup.Spec.Status.ClusterStatus == ""
+}
+
 // isUpgradeDone returns true if an upgrade has already been fully processed and has
 // the status of either Successed or Failed
 func isUpgradeDone(cup *provisioncsv3.ClusterUpgrade) bool {
```
Comment (ashleyschuett) [inline pkg/coordinator/upgrade_controller.go]: is this used any more?

File: pkg/coordinator/upgrade_controller.go (hunk 2 above)
Comment (mattkelly) [reply pkg/coordinator/upgrade_controller.go]: Fixed

File: pkg/coordinator/upgrade_controller.go (hunk 3 above)
Comment (mattkelly) [reply pkg/coordinator/upgrade_controller.go]: Yup! The sourcegraph browser plugin is great for checking stuff like this on PRs btw :)

**Pull request merged**